Sources/Build/Buildable.swift

```diff
@@ -9,20 +9,43 @@
 */
 
 import PackageType
+import struct Utility.Path
 
 protocol Buildable {
     var targetName: String { get }
     var isTest: Bool { get }
 }
 
+extension CModule {
+    func workingDirectory(prefix: String) -> String {
+        return Path.join(prefix, "\(c99name).build")
+    }
+}
+
 extension Module: Buildable {
     var isTest: Bool {
         return self is TestModule
     }
 
-    var Xcc: [String] {
+    func XccFlagsForPrefix(prefix: String) -> [String] {
         return recursiveDependencies.flatMap { module -> [String] in
-            if let module = module as? CModule {
+            if let module = module as? ClangModule {
+                var moduleMap: String? = nil
+                
+                if module.moduleMapPath.isFile {
+                    moduleMap = module.moduleMapPath
+                }
+    
+                let genModuleMap = Path.join(module.workingDirectory(prefix), module.moduleMap)
+                if genModuleMap.isFile {
+                    moduleMap = genModuleMap
+                }
+                //No module map found, return with no args
```

Comment:
This is a confusing comment, given the code then does the opposite. I would probably err on duplicating the `-Xcc` so this code can use an early return on line 36, or else factoring out the "get the module map path and/or generate" piece from the "generate the -Xcc args" part (again so an early return can be used).